sdk/monitor/monitor-query/src/logsQueryClient.ts

```diff
@@ -115,11 +115,23 @@ export class LogsQueryClient {
 
     const parsedBody = JSON.parse(rawResponse.bodyAsText!);
     flatResponse.tables = parsedBody.tables;
-    return {
+    const result: LogsQueryResult = {
       tables: flatResponse.tables.map(convertGeneratedTable),
       statistics: flatResponse.statistics,
-      visualization: flatResponse.render
+      visualization: flatResponse.render,
+      error: flatResponse.error
     };
+    if (result.tables && result.error) {
+      result.status = "Partial";
+    } else if (result.error && !result.tables) {
+      result.status = "Failed";
+    } else {
+      result.status = "Success";
+    }
```

Comment:
That's interesting. In a different part of your code, you have `Failed` as the fall-through case, but here it's success. Any reason for the difference?

Reply:
No reason really.  But now I am thinking if i should refine this - since the tables is never optional in the single query response.

Reply:
```ts
    if (!result.error) {
      // if there is no error field, it is success
      result.status = "Success";
    } else {
      // result.tables is always present in single query response, even is there is error
      if (result.tables.length === 0) {
        result.status = "Failed";
      } else {
        result.status = "Partial";
      }
    }
    ```

Reply:
How about this? So in the other batch query response - the tables can be undefined, but in single query response the "tables" field is always returned. So I think the above code snippet in the comment seems more appropriate for single query.

Reply:
@chradek  wanted follow up on this